### ISSUE TITLE
question: do I need to provide defaults when overriding exclude list?

### ISSUE THREAD

**Issue by Integralist** [zakharykaplan/nvim-retrail#6]
I'm doing the following, is this necessary or can I just add the one filetype and it'll get _appended_ to the defaults?

Thanks!

```lua
    use { "kaplanz/nvim-retrail",
      config = function()
        require("retrail").setup({
          filetype = {
            exclude = {
              "markdown",
              -- following are the defaults I've copied over in case they get overwritten
              "",
              "alpha",
              "checkhealth",
              "diff",
              "help",
              "lspinfo",
              "man",
              "mason",
              "TelescopePrompt",
              "Trouble",
              "WhichKey",
            }
          }
        })
      end
    }
```

**Comment by kaplanz** [comment]
Great question!

When designing the configuration I basically had to make this exact design choice. I ended up opting for the user's config to **override** the defaults rather than append, my reasoning being that otherwise there'd be no (obvious) way to unset defaults. (Although, as an aside `exclude`s have priority over `include`s, so that could have be used to partially override defaults.)

TL;DR how you currently have it set up is correct.

EDIT: At some future point I would consider adding an option such as `filetype.append` to enable such behaviour in a backwards compatible way. I would appreciate any feedback or discussion if that would be useful to users.

**Comment by Integralist** [comment]
Awesome, thanks for the detailed answer 👍🏻 

I feel like `filetype.append` would be nice to have because I generally trust a plugin author has done the hard work of configuring sensible defaults and in cases like this plugin, I only need to 'tweak' one thing. So I'm a big 👍🏻 on supporting this.

Thanks again!

**Comment by JacobHayes** [comment]
For anyone else wanting to extend the include/excludes (ie: not overwrite the existing defaults), I was able to use this w/ `lazy.nvim`:

```lua
{
  -- Highlights trailing spaces + deletes on write
  "kaplanz/retrail.nvim",
  opts = function(_, opts)
    local opts = require("retrail.config").parse(opts)

    opts.enabled.filetype.dashboard = false -- `include` and `exclude` are resolved into `enabled` 
    vim.list_extend(opts.filetype.exclude, { "dashboard" }) -- not used after retrail.config.parse(...), but being cautious
    return opts
  end,
}
```

Definitely not "simpler", but at least any additions to the excludes upstream will be added when you updated the plugin. 😅

**Comment by kaplanz** [comment]
Great comment.

As an aside after working through this, do you have any opinions on how a non-breaking config change to update (without overriding) the default `include` and `exclude` lists should look?

I never got around to adding `filetype.append` since I didn't feel I could get the design right while staying simple and discoverable. IIRC I played around with the idea of having a separate option `override` which defaults to `true` and when `false` causes both lists to append rather than override, but was worried that wouldn't be so obvious how it's used.
